### PR TITLE
Separate internal/external buses for sw-emulator mailbox to support uC -> SoC transactions

### PR DESCRIPTION
**Pull request by rusty1968**
-  Mailbox split into Soc and Caliptra side.
- State machine implements ExecSoc and ExecUc states and respective transition arcs.
- Implementation of unlock register.
- Integration tests

### CODE REVIEW

File: drivers/src/mailbox.rs [hunk 1]
```diff
@@ -72,6 +72,7 @@ impl Mailbox {
         let mbox = mbox::RegisterBlock::mbox_csr();
         match mbox.status().read().mbox_fsm_ps() {
             MboxFsmE::MboxExecuteUc => Some(MailboxRecvTxn::default()),
+            MboxFsmE::MboxExecuteSoc => Some(MailboxRecvTxn::default()),
```
Comment (korran) [inline drivers/src/mailbox.rs]: Why is this necessary? If I remove this line, all the tests pass.

File: sw-emulator/lib/periph/src/mailbox.rs [hunk 2]
```diff
@@ -86,154 +95,337 @@ impl Default for MailboxRam {
...
-    pub fn new(ram: MailboxRam) -> Self {
-        Self {
-            regs: Rc::new(RefCell::new(MailboxRegs::new(ram))),
-        }
+impl MailboxExternal {
+    pub fn new(regs: Soc2CaliptraMailboxRegs) -> Self {
+        Self { regs }
     }
-
     pub fn read_dlen(&mut self) -> Result<u32, BusError> {
```
Comment (korran) [inline sw-emulator/lib/periph/src/mailbox.rs:113]: Who calls these MailboxExternal functions? If I delete them all the tests still pass.

File: sw-emulator/lib/periph/src/mailbox.rs [hunk 3]
```diff
@@ -297,14 +491,18 @@ impl MailboxRegs {
             execute: ReadWriteRegister::new(Self::EXEC_VAL),
             _status: ReadWriteRegister::new(Self::STATUS_VAL),
             state_machine: StateMachine::new(Context::new(ram)),
+            requester: MailboxRequester::Caliptra,
         }
     }
+    pub fn request(&mut self, requester: MailboxRequester) {
```
Comment (korran) [inline sw-emulator/lib/periph/src/mailbox.rs:510]: set_requester()?

File: drivers/src/mailbox.rs (hunk 1 above)
Comment (rusty1968) [reply drivers/src/mailbox.rs]: it is not there anymore. this is outdated.

File: sw-emulator/lib/periph/src/mailbox.rs (hunk 2 above)
Comment (rusty1968) [reply sw-emulator/lib/periph/src/mailbox.rs:113]: sure - will remove. Thanks!

File: sw-emulator/lib/periph/src/mailbox.rs (hunk 3 above)
Comment (rusty1968) [reply sw-emulator/lib/periph/src/mailbox.rs:510]: yep - will change.

File: sw-emulator/lib/periph/src/mailbox.rs (hunk 2 above)
Comment (rusty1968) [reply sw-emulator/lib/periph/src/mailbox.rs]: Done.

**Review by korran — APPROVED** [review]
